### PR TITLE
Adding AuthorizedOrgsDesc to AccessContextManager

### PR DESCRIPTION
**Pull request by vmoros**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_access_context_manager_authorized_orgs_desc`
```

### CODE REVIEW

File: mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+# limitations under the License.
+--- !ruby/object:Api::Resource
+  name: 'AuthorizedOrgsDesc'
+  base_url: ""
+  create_url: "{{parent}}/authorizedOrgsDescs"
+  self_link: "{{name}}"
+  update_verb: :PATCH
+  references: !ruby/object:Api::Resource::ReferenceLinks
+    guides:
+      'gcloud docs': 'https://cloud.google.com/beyondcorp-enterprise/docs/cross-org-authorization'
```
Comment (drebes) [inline mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml:21]: This link 404s.

Comment (vmoros) [reply mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml:21]: Yup, that's where the guide will eventually be but it's not there yet. Currently it's only available internally here: https://cloud.devsite.corp.google.com/beyondcorp-enterprise/docs/cross-org-authorization

Should I remove the link to the guide until it's available?

**Comment by vmoros** [comment]
@shuyama1, the most recent VCR tests are failing with a bunch of HTTP 409 errors that say `ALREADY_EXISTS`. I'm guessing a previous run of tests did not clean up properly. How can I fix that?

**Comment by shuyama1** [comment]
Cleaned up resource. Kicking off a run to see if it works.
/gcbrun

File: mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml (hunk 1 above)
Comment (vmoros) [reply mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml:21]: The user guide is now live at the URL in the code so it no longer 404s.

**Comment by shuyama1** [comment]
cleaned up `TestAccAccessContextManager` VCR cassette and kick off a rerun to make sure the modified test passed in VCR. What's expected: `TestAccAccessContextManager` failed in replaying, rerun in recording and then passed.

/gcbrun

**Comment by shuyama1** [comment]
Triggered an independent run on CI. Test result:
```
------- Stdout: -------
=== RUN   TestAccAccessContextManager
=== RUN   TestAccAccessContextManager/service_perimeter
=== RUN   TestAccAccessContextManager/service_perimeter_update
=== RUN   TestAccAccessContextManager/service_perimeter_resource
=== RUN   TestAccAccessContextManager/access_level
=== RUN   TestAccAccessContextManager/access_level_full
=== RUN   TestAccAccessContextManager/access_level_custom
=== RUN   TestAccAccessContextManager/gcp_user_access_binding
=== PAUSE TestAccAccessContextManager/gcp_user_access_binding
=== RUN   TestAccAccessContextManager/access_policy_scoped
=== RUN   TestAccAccessContextManager/access_levels
=== RUN   TestAccAccessContextManager/access_level_condition
=== RUN   TestAccAccessContextManager/service_perimeters
=== RUN   TestAccAccessContextManager/authorized_orgs_desc
provider_test.go:315: Step 1/2 error: Error running apply: exit status 1
2023/02/21 18:21:36 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.17.3
Error: Error when reading or editing AccessContextManagerAuthorizedOrgsDesc "accessPolicies/675291891979/authorizedOrgsDescs/fakeDescName": googleapi: Error 403: The caller does not have permission
on terraform_plugin_test.tf line 3, in resource "google_access_context_manager_authorized_orgs_desc" "authorized-orgs-desc":
3: resource "google_access_context_manager_authorized_orgs_desc" "authorized-orgs-desc" {
=== RUN   TestAccAccessContextManager/access_policy
=== CONT  TestAccAccessContextManager/gcp_user_access_binding
--- FAIL: TestAccAccessContextManager (327.80s)
--- PASS: TestAccAccessContextManager/service_perimeter (16.28s)
--- PASS: TestAccAccessContextManager/service_perimeter_update (48.43s)
--- PASS: TestAccAccessContextManager/service_perimeter_resource (24.32s)
--- PASS: TestAccAccessContextManager/access_level (22.22s)
--- PASS: TestAccAccessContextManager/access_level_full (14.00s)
--- PASS: TestAccAccessContextManager/access_level_custom (13.38s)
--- PASS: TestAccAccessContextManager/access_policy_scoped (11.40s)
--- PASS: TestAccAccessContextManager/access_levels (32.67s)
--- PASS: TestAccAccessContextManager/access_level_condition (26.07s)
--- PASS: TestAccAccessContextManager/service_perimeters (32.69s)
--- FAIL: TestAccAccessContextManager/authorized_orgs_desc (68.19s)
--- PASS: TestAccAccessContextManager/access_policy (18.14s)
--- PASS: TestAccAccessContextManager/gcp_user_access_binding (24.87s)
FAIL
```

File: mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func testAccAccessContextManagerAuthorizedOrgsDesc_basicTest(t *testing.T) {
+	context := map[string]interface{}{
+		"org_id": getTestOrgFromEnv(t),
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go:14]:
```suggestion
		"org_id": GetTestOrgFromEnv(t),
```

File: mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	vcrTest(t, resource.TestCase{
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go:17]:
```suggestion
	VcrTest(t, resource.TestCase{
```

File: mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go:19]:
```suggestion
		Providers:    TestAccProviders,
```

File: mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+	return func(s *terraform.State) error {
+		for name, rs := range s.RootModule().Resources {
+			if rs.Type != "google_access_context_manager_authorized_orgs_desc" {
+				continue
+			}
+			if strings.HasPrefix(name, "data.") {
+				continue
+			}
+
+			config := googleProviderConfig(t)
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go:63]:
```suggestion
			config := GoogleProviderConfig(t)
```

File: mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+				return err
+			}
+
+			billingProject := ""
+
+			if config.BillingProject != "" {
+				billingProject = config.BillingProject
+			}
+
+			_, err = sendRequest(config, "GET", billingProject, url, config.userAgent, nil)
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go:76]:
```suggestion
			_, err = SendRequest(config, "GET", billingProject, url, config.userAgent, nil)
```

**Review by shuyama1 — CHANGES_REQUESTED** [review]
A recent change to MM capitalize multiple utility functions to make them public, and that's probably why the build failed for this PR. Sorry about it!

File: mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+				return err
+			}
+
+			billingProject := ""
+
+			if config.BillingProject != "" {
+				billingProject = config.BillingProject
+			}
+
+			_, err = SendRequest(config, "GET", billingProject, url, config.userAgent, nil)
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_access_context_manager_authorized_orgs_desc_test.go:76]:
```suggestion
			_, err = SendRequest(config, "GET", billingProject, url, config.UserAgent, nil)
```

**Review by shuyama1 — COMMENTED** [review]
sorry for missing this one!

File: mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        - :AUTHORIZATION_DIRECTION_TO
+        - :AUTHORIZATION_DIRECTION_FROM
+    - !ruby/object:Api::Type::Enum
+      name: authorizationType
+      input: true
+      description: |
+        A granular control type for authorization levels. Valid value is "AUTHORIZATION_TYPE_TRUST".
+      values:
+        - :AUTHORIZATION_TYPE_UNSPECIFIED
+        - :AUTHORIZATION_TYPE_TRUST
```
Comment (shuyama1) [inline mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml:119]: Is there a server-side default values for these fields -- Does API return values for them if they are not explicitly specified?

Also should we remove `*_UNSPECIFIED` values? My understanding is that they are not valid values for users to specify.

**Review by shuyama1 — COMMENTED** [review]
Only a couple small comments.

I've triggered VCR recording test on TC and therefore cassette should be forced recording for the newly added test

File: mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml (hunk 8 above)
Comment (vmoros) [reply mmv1/products/accesscontextmanager/AuthorizedOrgsDesc.yaml:119]: There are no server-side default values for `assetType`, `authorizationDirection`, or `authorizationType`. The user must specify them.

The `*_UNSPECIFIED` values exist under the hood, which is why I added them in the code here, but they cannot be used so IMO it's reasonable to remove them here.

**Review by shuyama1 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**